homeassistant/components/powerwall/sensor.py

```diff
@@ -0,0 +1,159 @@
+"""Support for August sensors."""
+import logging
+
+from homeassistant.const import DEVICE_CLASS_BATTERY, DEVICE_CLASS_POWER
+
+from . import PowerWallEntity
+from .const import (
+    ATTR_ENERGY_EXPORTED,
+    ATTR_ENERGY_IMPORTED,
+    ATTR_FREQUENCY,
+    ATTR_INSTANT_AVERAGE_VOLTAGE,
+    DOMAIN,
+    POWERWALL_API_CHARGE,
+    POWERWALL_API_METERS,
+    POWERWALL_COORDINATOR,
+    POWERWALL_IP_ADDRESS,
+    POWERWALL_SITE_INFO,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up the August sensors."""
+    powerwall_data = hass.data[DOMAIN][config_entry.entry_id]
+    _LOGGER.debug("Powerwall_data: %s", powerwall_data)
+
+    coordinator = powerwall_data[POWERWALL_COORDINATOR]
+    site_info = powerwall_data[POWERWALL_SITE_INFO]
+    ip_address = powerwall_data[POWERWALL_IP_ADDRESS]
+
+    entities = []
+    for meter in coordinator.data[POWERWALL_API_METERS]:
+        entities.append(
+            PowerWallEnergySensor(meter, coordinator, site_info, ip_address)
+        )
+
+    entities.append(PowerWallChargeSensor(coordinator, site_info, ip_address))
+
+    async_add_entities(entities, True)
+
+
+class PowerWallSensor(PowerWallEntity):
+    """Base class for powerwall sensors."""
+
+    def __init__(self, coordinator, site_info, ip_address):
+        """Initialize the sensor."""
+        super().__init__(coordinator, site_info, ip_address)
+        self._coordinator = coordinator
+
+    @property
+    def available(self):
```

Comment:
Can we move all of the common entity properties of binary sensor and sensor platform entities to `PowerWallEntity`?

Reply:
Yes done